samples/dynamic_alloc/main.cpp

```diff
@@ -0,0 +1,177 @@
+//////////////////////////////////////////////////////////////////////////////
+//
+//  Detours Test Program
+//
+//  Microsoft Research Detours Package
+//
+//  Copyright (c) Microsoft Corporation.  All rights reserved.
+//
+#define _CRT_SECURE_NO_WARNINGS
+#include <iostream>
+#include <functional>
+#include <assert.h>
+#include <windows.h>
+#include <detours.h>
+
+extern "C" {
+  void *CodeTemplate();
+  void *CodeTemplate_End();
+}
+
+void Log(LPCTSTR format, ...) {
+  TCHAR linebuf[1024];
+  va_list v;
+  va_start(v, format);
+  wvsprintf(linebuf, format, v);
+  va_end(v);
+  OutputDebugString(linebuf);
+}
+
+void *target_function() {
+  std::cout << '+' << __FUNCTION__ << std::endl;
+  return nullptr;
+}
+
+bool DetourTransaction(std::function<bool()> callback) {
+  LONG status = DetourTransactionBegin();
+  if (status != NO_ERROR) {
+    Log("DetourTransactionBegin failed with %08x\n", status);
+    return status;
+  }
+
+  if (callback()) {
+    status = DetourTransactionCommit();
+    if (status != NO_ERROR) {
+      Log("DetourTransactionCommit failed with %08x\n", status);
+    }
+  }
+  else {
+    status = DetourTransactionAbort();
+    if (status == NO_ERROR) {
+      Log("Aborted transaction.\n");
+    }
+    else {
+      Log("DetourTransactionAbort failed with %08x\n", status);
+    }
+  }
+  return status == NO_ERROR;
+}
+
+class CodeRegionFactory final {
+  template <typename T>
+  static const T *at(const void *base, uint32_t offset) {
+    return
+      reinterpret_cast<const T*>(
+        reinterpret_cast<const uint8_t*>(base) + offset);
+  }
+
+  template <typename T>
+  static T *at(void *base, uint32_t offset) {
+    return
+      reinterpret_cast<T*>(
+        reinterpret_cast<uint8_t*>(base) + offset);
+  }
+
+  void *region_{};
```

Comment:
Please avoid the C++ {} style initializers in the sample code.  There are a few other places where you've used that style.

Reply:
Thank you for the comments.  I added more comments and replaced initialization code.